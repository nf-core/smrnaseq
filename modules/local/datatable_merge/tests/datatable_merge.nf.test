nextflow_process {

    name "Test Process DATATABLE_MERGE"
    script "../main.nf"
    process "DATATABLE_MERGE"
    tag "modules"
    tag "modules_local"
    tag "datatable_merge"

    test("Contains hsa-miR-365b-3p, hsa-miR-7-5p, hsa-miR-103a-3p") {

        when {
            params {
                outdir = "${outputDir}"
            }
            process {
                """
                input[0] = [[],file("https://github.com/nf-core/test-datasets/raw/smrnaseq/nf-test_data/datatable_merge/small_mirtop_dataset.txt", checkIfExists: true)]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()

            with(process.out.mirna_tsv) {
                with(get(0)) {
                    assert get(0).endsWith(".tsv")

                    // Check for specific miRNAs
                    def lines = path(get(0)).readLines()
                    assert lines.any { it.contains("hsa-miR-365b-3p") }
                    assert lines.any { it.contains("hsa-miR-7-5p") }
                    assert lines.any { it.contains("hsa-miR-103a-3p") }
                }
            }
        }
    }

    test("Does not contain hsa-miR-107, hsa-miR-365a-3p") {

        when {
            params {
                outdir = "${outputDir}"
            }
            process {
                """
                input[0] = [[],file("https://github.com/nf-core/test-datasets/raw/smrnaseq/nf-test_data/datatable_merge/small_mirtop_dataset.txt", checkIfExists: true)]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()

            with(process.out.mirna_tsv) {
                with(get(0)) {
                    assert get(0).endsWith(".tsv")

                    // Check for the absence of specific miRNAs
                    def lines = path(get(0)).readLines()
                    assert !lines.any { it.contains("hsa-miR-107") }
                    assert !lines.any { it.contains("hsa-miR-365a-3p") }
                }
            }
        }
    }

}
