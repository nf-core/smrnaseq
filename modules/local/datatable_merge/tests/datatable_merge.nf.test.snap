{
    "Contains hsa-miR-365b-3p, hsa-miR-7-5p, hsa-miR-103a-3p": {
        "content": [
            {
                "0": [
                    "mirna.tsv:md5,f59a6aeb15588c43c2977950a1b0a080"
                ],
                "1": [
                    "versions.yml:md5,13bf3c8bbf1285dfc0ef547dcbb692b2"
                ],
                "mirna_tsv": [
                    "mirna.tsv:md5,f59a6aeb15588c43c2977950a1b0a080"
                ],
                "versions": [
                    "versions.yml:md5,13bf3c8bbf1285dfc0ef547dcbb692b2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-30T12:57:47.129770995"
    },
    "Does not contain hsa-miR-107, hsa-miR-365a-3p": {
        "content": [
            {
                "0": [
                    "mirna.tsv:md5,f59a6aeb15588c43c2977950a1b0a080"
                ],
                "1": [
                    "versions.yml:md5,13bf3c8bbf1285dfc0ef547dcbb692b2"
                ],
                "mirna_tsv": [
                    "mirna.tsv:md5,f59a6aeb15588c43c2977950a1b0a080"
                ],
                "versions": [
                    "versions.yml:md5,13bf3c8bbf1285dfc0ef547dcbb692b2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-30T12:57:56.990602055"
    }
}